### PR TITLE
Add context menu directive

### LINKED ISSUE #9926: An error in console: Cannot read properties of null (reading 'deactivate')
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues?q=is%3Aissue) that already describes my problem.

### Describe the Bug

When I enter a collection and then go to a specific record and then (after changes or without them) I leave record via the back button, everything is ok. However, if I am in the record (without even making any changes), I click on the next collection skipping the back button. I get an error in the console. 

### To Reproduce

Then go to the collection and then to the specific record. Exit it but by clicking directly on another collection (not via the back button). Check your console. 

### Errors Shown
        
       TypeError: Cannot read properties of null (reading 'deactivate')

### What version of Directus are you using?

9.0.1

### What version of Node.js are you using?

16.13.0

### What database are you using?

Sql server 2017

### What browser are you using?

Edge 95.0.1020.53

### What operating system are you using?

Debian 11.1

### How are you deploying Directus?

locally

### PR DESCRIPTION
**Pull request by azrikahar**
Fixes #9926

Ref: https://github.com/directus/directus/pull/9271#discussion_r740579195

---

Might have butchered this implementation since I've never personally added custom directives, but hope it's in the right direction 😅

### CODE REVIEW

File: app/src/directives/context-menu/context-menu.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import { Directive, DirectiveBinding } from 'vue';
+
+function mounted(element: HTMLElement, binding: DirectiveBinding): void {
+	const contextMenu = binding.instance?.$refs['contextMenu'];
```
Comment (rijkvanzanten) [inline app/src/directives/context-menu/context-menu.ts:4]: I don't think we can make this automated 🤔 There's gonna be cases where theres's more than one context menu in the same parent component. I think we should update this so the contextMenu ref is pulled in from the binding, which then means the use becomes:

`<element v-context-menu="contextMenuRef">`

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]

**Comment by rijkvanzanten** [comment]
Other than that binding, this is looking good!

File: app/src/directives/context-menu/context-menu.ts (hunk 1 above)
Comment (azrikahar) [reply app/src/directives/context-menu/context-menu.ts]: > There's gonna be cases where theres's more than one context menu in the same parent component. 

That's true 😄 Updated the PR with this change.

**Comment by azrikahar** [comment]
@joselcvarela thanks for fixing the empty context menu. That's a tricky one, assuming there's more items added to the list so the condition will need to be updated 🤔 but it'll have to do for now I guess.

**Comment by joselcvarela** [comment]
Hey @azrikahar 
Just made a few more changes, check it out.
Basically:
- if `v-contextmenu` is attached, always listen to context menu in order to prevent browser context menu. This is useful when we want to attach a context menu but the `v-menu` has a condition.
- only `Show hidden collections` if there is hidden collections. The fix above is used here (because `v-menu` will not render)
- the arrow on `v-menu` was annoying me for a while because it was on top of items in `v-menu` so a little part was shown. Not anymore!!! 😅

**Comment by rijkvanzanten** [comment]
> the arrow on v-menu was annoying me for a while because it was on top of items in v-menu so a little part was shown. Not anymore!!! 😅

I'm sure @benhaynes will appreciate this

**Review by rijkvanzanten — APPROVED** [review]

**Comment by benhaynes** [comment]
You know it!! Thanks, @joselcvarela!! ❤️

**Pull request merged**